server/src/app.ts

```diff
@@ -82,17 +78,33 @@ export const main = async (app: Express) => {
           },
         },
       },
+      update: {},
     });
   }
 
+  async function findOrCreateUser(email: string) {
+    try {
+      return await upsertUser(email);
+    } catch (e) {
+      if (
+        e instanceof Prisma.PrismaClientKnownRequestError &&
+        e.code === 'P2002'
```

Comment:
This error code has named variable in the `server/src/prisma.ts` https://github.com/freeCodeCamp/chapter/blob/bcd226b9bf5f0b14e749690c0d5a37fb0058bb74/server/src/prisma.ts#L10